oraclizeAPI_0.4.25.sol

```diff
@@ -366,155 +420,1286 @@ contract usingOraclize {
+    {
+        return oraclize.convertToERC20Price(
+            oraclize.getPrice(_datasource, _gasLimit),
+            _tokenAddress
+        );
+    }
+
+    function oraclize_getPriceERC20(
+        string memory _datasource,
+        address _address
+    )
+        oraclizeAPI
+        internal
+        returns (uint256 _queryPrice)
+    {
+        return oraclize.convertToERC20Price(
+            oraclize.getPrice(_datasource, _address),
+            oraclize.addressCustomPaymentToken(address(this))
+        );
+    }
+
+    function oraclize_getPriceERC20(
+        byte _datasource,
+        address _address
+    )
+        oraclizeAPI
+        internal
+        returns (uint256 _queryPrice)
+    {
+        return oraclize.convertToERC20Price(
+            oraclize.getPrice(_datasource, _address),
+            oraclize.addressCustomPaymentToken(address(this))
+        );
+    }
+
+    function oraclize_getPriceERC20(
+        string memory _datasource,
+        address _address,
+        address _tokenAddress
+    )
+        oraclizeAPI
+        internal
+        returns (uint256 _queryPrice)
+    {
+        return oraclize.convertToERC20Price(
+            oraclize.getPrice(_datasource, _address),
+            _tokenAddress
+        );
+    }
+
+    function oraclize_getPriceERC20(
+        byte _datasource,
+        address _address,
+        address _tokenAddress
+    )
+        oraclizeAPI
+        internal
+        returns (uint256 _queryPrice)
+    {
+        return oraclize.convertToERC20Price(
+            oraclize.getPrice(_datasource, _address),
+            _tokenAddress
+        );
+    }
+
+    function oraclize_getPriceERC20(
+        string memory _datasource,
+        address _address,
+        uint256 _gasLimit
+    )
+        oraclizeAPI
+        internal
+        returns (uint256 _queryPrice)
+    {
+        return oraclize.convertToERC20Price(
+            oraclize.getPrice(_datasource, _gasLimit, _address),
+            oraclize.addressCustomPaymentToken(address(this))
+        );
+    }
+
+    function oraclize_getPriceERC20(
+        byte _datasource,
+        address _address,
+        uint256 _gasLimit
+    )
+        oraclizeAPI
+        internal
+        returns (uint256 _queryPrice)
+    {
+        return oraclize.convertToERC20Price(
+            oraclize.getPrice(_datasource, _gasLimit, _address),
+            oraclize.addressCustomPaymentToken(address(this))
+        );
+    }
+
+    function oraclize_getPriceERC20(
+        string memory _datasource,
+        address _address,
+        uint256 _gasLimit,
+        address _tokenAddress
+    )
+        oraclizeAPI
+        internal
+        returns (uint256 _queryPrice)
+    {
+        return oraclize.convertToERC20Price(
+            oraclize.getPrice(_datasource, _gasLimit, _address),
+            _tokenAddress
+        );
+    }
+
+    function oraclize_getPriceERC20(
+        byte _datasource,
+        address _address,
```

Comment:
@gskapka was saying _address is to vague, maybe something more specific is needed

Reply:
Change(s) made and functions much clearer now!